doc/specs/#1595 - Suggestions UI/Snippets.md

```diff
@@ -0,0 +1,588 @@
+load actions from that file as if it were a fragment extension as well. That
+will start with the `startingDirectory` for any new panes created. If the user
+has shell integration configured to tell the Terminal about the CWD, then we'll
+refresh that list as the user changes directories.
+
+* In `Terminal.Settings.Model`, we will store a cached map of path->actions.
+  * that if multiple panes are all in the same CWD, they don't need to
+    individually re-read the file from disk and regenerate that part of the map.
+* I believe it should be impossible for a keybinding to be bound to a local
+  action. Even if it has an ID, the file won't be loaded when we build the
+  keymap, and we don't really want the keymap changing based on CWD. Also, with
+  the actions living in an entirely separate map of CWD->actions, the
+  keybindings in the main map won't be able to easily get to them. See also
+  [Security considerations](#security) for more.
+* If the Snippets pane or Suggestions UI is opened with `snippets` as a source,
+  then we'll just append the appropriate list of suggestions for the active
+  control's CWD.
+  * We don't need to have the control raise an event when the CWD changes - we
+    can lazy-load these actions when a UI element that requires it is first
+    invoked.
+* The Command Palette is trickier, since it binds directly to the action map.
+  We'd need to be able to freely modify that map at runtime, which might be
+  prohibitively annoying. <!-- * If we want these actions to show up in the
+  Command Palette, we'll need to:
+  * We'll stash these actions in the action map as they're loaded. (in
+    `Terminal.Settings.Model`)
+  * We'll need to be able to dynamically remove them at runtime from the map (in
+    `Terminal.Settings.Model`) -->
+* If we find multiple `.wt.json` files in the ancestors of the CWD (e.g. for
+  `c:\a\b\c\d\`, there's a `c:\a\.wt.json` and a `c:\a\b\c\.wt.json`), then
+  we'll add each one separately to the map of paths->directories. When
+  requesting the actual actions for `c:\a\b\c\d\`, we'll layer the ones from
+  `c:\a\` before the ones from `c:\a\b\c`, so that deeper descendants take
+  precedence.
+
+### Saving snippets from the commandline
+
+_This has already been prototyped in [#16513]_
+
+Users should be able to save commands as snippets directly from the commandline.
+Consider: you've just run the command that worked the way you need it to. You
+shouldn't have to open the settings to then separately copy-paste the command in
+to save it. It should be as easy as <kbd>Up</kbd>, <kbd>Home</kbd>, `wt save `,
+<kbd>Enter</kbd>.
+
+The exact syntax as follows:
+
+This will be powered by a `saveSnippet` action behind the scenes. After running
+this command, a toast will be presented to the user to indicate success/failure.
+The schema for this action is as follows:
+
+```json
+"SaveSnippetAction": {
+  "description": "Arguments corresponding to a saveSnippet Action",

```

Comment:
I don't think this needs to be parseable from JSON. It's an implementation detail.